manager/allocator/network.go

```diff
@@ -641,6 +644,35 @@ func (a *Allocator) commitAllocatedNode(ctx context.Context, batch *store.Batch,
 	return nil
 }
 
+func updatePortsInHostPublishMode(s *api.Service) {
+	if s.Endpoint != nil {
+		var portConfigs []*api.PortConfig
+		for _, portConfig := range s.Endpoint.Ports {
+			if portConfig.PublishMode == api.PublishModeIngress {
+				portConfigs = append(portConfigs, portConfig)
+			}
+		}
+		s.Endpoint.Ports = portConfigs
+	}
+
+	if s.Spec.Endpoint != nil {
+		if s.Endpoint == nil {
+			s.Endpoint = &api.Endpoint{}
+		}
+		if s.Endpoint.Spec == nil {
+			s.Endpoint.Spec = &api.EndpointSpec{}
+		}
+		s.Endpoint.Spec.Ports = nil
+		for _, portConfig := range s.Spec.Endpoint.Ports {
+			s.Endpoint.Spec.Ports = append(s.Endpoint.Spec.Ports, portConfig.Copy())
```

Comment:
Actually, I guess this is effectively making an exact copy (unless the mode changed somehow), but `s.Endpoint.Spec = s.Spec.Endpoint.Copy()` seems like a simpler/cleaner way to do that.